src/api/goods.ts

```diff
@@ -5,15 +5,28 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
 
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL)
-    .then(response => response.json());
+    .then(response => {
+      if (!response.ok) {
+        throw new Error('Not your day... Please try again later.');
+      }
+
+      return response.json();
+    });
 }
 
-export const get5First = () => {
+export const get5First = (): Promise<Good[]> => {
   return getAll()
-    .then(goods => goods); // sort and get the first 5
+    .then(goods => goods)
+    .then((response) => {
+      const sortedGoods = response.sort((a: Good, b: Good) => (
+        a.name.localeCompare(b.name)));
+
+      return sortedGoods.splice(0, 5);
+    });
 };
 
-export const getRedGoods = () => {
+export const getRedGoods = (): Promise<Good[]> => {
   return getAll()
-    .then(goods => goods); // get only red
+    .then(goods => goods)
```

Comment:
looks like redundant then

Reply:
@s3rg4sh remove this `then` or explain why u need it pls